modules/renter/repair.go

```diff
@@ -150,6 +151,21 @@ func (f *file) chunkHosts(chunk uint64) []modules.NetAddress {
 	return old
 }
 
+// expiringContracts returns the contracts that will expire soon.
+// TODO: what if contract has fully expired?
```

Comment:
Hmm. Right now you can renew it as long as it is in the source location, but if is gone both locally and remotely, you should probably just leave it and let the UI figure out how to report "not repairing" + "0 blocks left"